File: js/sliding_window.js
let slidingWindow = function(vec, window) {
    let val = vec.slice(0, window).reduce((a, c) => a + c, 0) / window,
        newVec = [val];
    for (let i = 0; i < vec.length - window - 1; i++) {
        val += (vec[i + window] - vec[i]) / window;
        newVec.push(val);
    };
    return newVec
}

File: python/slidingWindow.py
# Copy of sliding_window from plotter
def sliding_window(xdomain, occupancy, window):
    occupancy_val = sum(occupancy[:window]) / window 
    print("length: " + str(xdomain))

    new_xdomain = [(xdomain[0] + xdomain[window - 1]) / 2]
    new_occupancy = [occupancy_val]
    for i in range(len(occupancy) - window - 1):
        new_xdomain.append((xdomain[i + 1] + xdomain[i + window]) / 2)  # Added parentheses here
        occupancy_val += (occupancy[i + window] - occupancy[i]) / window
        new_occupancy.append(occupancy_val)
    return {"new_xdomain": new_xdomain, "new_occupancy": new_occupancy}